packages/did-ethr-resolver/src/functions/functions.ts

```diff
@@ -221,13 +218,13 @@ const getEventsFromBlock = (
   contractInterface: utils.Interface,
   address: string,
 ): Promise<unknown> => new Promise((resolve, reject) => {
-  const [, , identity] = did.split(':');
+  var identity = addressOf(did);
 
   provider.getLogs({
     address,
     fromBlock: block.toNumber(),
     toBlock: block.toNumber(),
-    topics: [null, `0x000000000000000000000000${identity.slice(2).toLowerCase()}`] as string[],
+    topics: [null, `0x000000000000000000000000${identity?.slice(2).toLowerCase()}`] as string[],
```

Comment:
rather than `?` here, what do you think of throwing an error if `identity` is `undefined` as that should really never happen

Reply:
Actually, it looks like `addressOf` is already throwing an error

Reply:
Actually `?` should be removed now, it was added when identity was obtained from the `split` and possibly could have been undefined.